server/services/inductionService.ts

```diff
@@ -8,20 +8,23 @@ export default class InductionService {
   constructor(private readonly educationAndWorkPlanClient: EducationAndWorkPlanClient) {}
 
   async inductionExists(prisonNumber: string, token: string): Promise<boolean> {
-    return (await this.getInduction(prisonNumber, token)) !== undefined
+    try {
+      await this.getInduction(prisonNumber, token)
+      return true
+    } catch (error) {
+      if (error.status === 404) {
+        return false
+      }
+      throw error
+    }
   }
 
   async getInduction(prisonNumber: string, token: string): Promise<InductionDto> {
     try {
       const inductionResponse = await this.educationAndWorkPlanClient.getInduction(prisonNumber, token)
       return toInductionDto(inductionResponse)
     } catch (error) {
-      if (error.status === 404) {
-        logger.info(`No Induction found for prisoner [${prisonNumber}] in Education And Work Plan API`)
-        return undefined
-      }
-
-      logger.error(`Error retrieving Induction data from Education And Work Plan: ${JSON.stringify(error)}`)
+      logger.error('Error retrieving Induction data from Education And Work Plan API', error)
```

Comment:
I had to make a minor change to `InductionService` in respect of what happens when the Induction API returns a 404.
Previously the code returned `undefined`, but I needed it to rethrow the exception so that calling code (the request handler) could act accordingly